packages/create-sitecore-jss/src/templates/nextjs-sxa/src/pages/api/sitemap.ts

```diff
@@ -0,0 +1,34 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import config from 'temp/config';
+import { GraphQLSitemapService } from '@sitecore-jss/sitecore-jss/site';
+
+const sitemapApi = async (req: NextApiRequest, res: NextApiResponse): Promise<void> => {
+  const { query: { id } } = req;
+  const currentUrl = process.env.PUBLIC_URL;
+  // create sitemap graphql service
+  const sitemapService = new GraphQLSitemapService({
+    endpoint: config.graphQLEndpoint,
+    apiKey: config.sitecoreApiKey,
+    siteName: config.jssAppName,
+  });
+
+  const existSitemap = await sitemapService.getExistsSitemap(id as string);
+  if (existSitemap) {
+    const sitemapResult = await fetch(`${currentUrl}${existSitemap}` as string, {
+      method: 'GET'
+    }).then(result => {
+      if (result.status === 200) {
+        return result.text();
+      }
+      return;
+    });
+  
```

Comment:
Need to set the 'Content-Type' header to xml somewhere ([example](https://github.com/Sitecore/jss/blob/poc/sxa-headless/samples/nextjs/src/pages/api/sitemap.ts#L20)). Also, should we gzip?